### PR TITLE
Feature/heatmap 2way sync

### PR DESCRIPTION
**Pull request by DipraAich**
### Description
Implementation of 2way sync for heatmap

### Issues Resolved
[#1011](https://github.com/opensearch-project/observability/issues/1011)

### CODE REVIEW

File: dashboards-observability/public/components/visualizations/charts/maps/heatmap.tsx [hunk 1]
```diff
@@ -37,7 +37,9 @@ export const HeatMap = ({ visualizations, layout, config }: any) => {
     isEmpty(zMetrics) ||
     isEmpty(data[xaxisField.label]) ||
     isEmpty(data[yaxisField.label]) ||
-    isEmpty(data[`${zMetrics.aggregation}(${zMetrics.name})`])
+    isEmpty(data[`${zMetrics.aggregation}(${zMetrics.name})`]) ||
+    dataConfig?.dimensions.length > 2 ||
+    dataConfig?.metrics.length > 1
     // indexOf(NUMERICAL_FIELDS, zMetrics.type) < 0
```
Comment (spattnaik) [inline dashboards-observability/public/components/visualizations/charts/maps/heatmap.tsx]: Do we need this commented line after the actual implementation?

Comment (DipraAich) [reply dashboards-observability/public/components/visualizations/charts/maps/heatmap.tsx]: Eliminated this commented code

File: dashboards-observability/public/components/visualizations/charts/maps/heatmap.tsx [hunk 2]
```diff
@@ -37,8 +37,9 @@ export const HeatMap = ({ visualizations, layout, config }: any) => {
     isEmpty(zMetrics) ||
     isEmpty(data[xaxisField.label]) ||
     isEmpty(data[yaxisField.label]) ||
-    isEmpty(data[`${zMetrics.aggregation}(${zMetrics.name})`])
-    // indexOf(NUMERICAL_FIELDS, zMetrics.type) < 0
+    isEmpty(data[`${zMetrics.aggregation}(${zMetrics.name})`]) ||
+    dataConfig?.dimensions.length > 2 ||
```
Comment (abasatwar) [inline dashboards-observability/public/components/visualizations/charts/maps/heatmap.tsx:41]: here do we need to add timestamp specific check?

**Review by abasatwar — CHANGES_REQUESTED** [review]

File: dashboards-observability/public/components/visualizations/charts/maps/heatmap.tsx (hunk 2 above)
Comment (DipraAich) [reply dashboards-observability/public/components/visualizations/charts/maps/heatmap.tsx:41]: If we do not add timestamp in heatmap it still renders if the metric and dimensions are proper. So, we do not mandatorily need timestamp check.

**Review by abasatwar — COMMENTED** [review]
looks good.

**Pull request closed without merging**